release/models/system/openconfig-hashing.yang

```diff
@@ -0,0 +1,336 @@
+            description "The transport layer fields that should be used to
+                compute the hash.";
+        }
+
+        uses ingress-interface-top;
+    }
+
+    grouping ingress-interface-top {
+        description "Top-level container for ingress subinterface.";
+        leaf ingress-interface {
+            type boolean;
+            description "Use the ingress subinterface identifired in the
+                calculation of the hash.";
+        }
+    }
+
+    grouping hashing-policy-config {
+        description
+          "Configuration data for hashing policies.";
+
+        leaf name {
+            type string;
+            description
+              "The name of the hashing policy.
+              When a configured user-controlled policy is created by the
+              system, it is instantiated with the same name in the
+              /system/hashing-policies/hashing-policy/name list.";
+        }
+
+        leaf seed {
+            type uint64;
+            description
+                "The seed used to initialize the hash algorithm";
+        }
+
+        leaf algorithm {
+            type string;
+            description
+                "The name of hash algorithm. This algorithm MUST
+                be a supported algorithm";
+        }
+    }
+
+    grouping hashing-policy-top {
+        description
+            "Top level grouping for hashing configuration and operational state
+            data.";
+
+        container hashing-policies {
+            description
+              "Top level container for hashing, including configuration and
+              state data.";
+
+            list hashing-policy {
+                key "name";
+
+                description
+                  "The list of named policies to be used on the device.";
+
+                leaf name {
+                    type leafref {
+                        path "../config/name";
+                    }
+                    description
+                      "References the name of the hashing policy.";
+                }
+                container config {
+                    description
+                      "Configurable items at the global hash policy level.";
+
+                    uses hashing-policy-config;
+                }
+                container state {
+                    config false;
+                    description
+                      "Operational state data at the global hash policy
+                      level.";
+
+                    uses hashing-policy-config;
+                }
+                container hash-field-modes {
+                    description
+                        "Container for specifying inputs to be used when
+                        calculating the hash.";
+                    container config {
+                        description
+                            "Configurable items at the hashing inputs level.";
+                        uses hashing-inputs;
+                    }
```

Comment:
It could be very useful to have separate hashing input fields for IPv4 vs. IPv6 packets given the additional entropy in the IPv6 flow label (e.g., L3 src and dest addresses+flow label for IPv6 vs. adding in the L4 src and dst ports for IPv4). If I understand correctly, this is not allowed per the current container structure. Any thoughts on revising to allow this additional flexibility?

Reply:
This is possible in the current model, one can configure the ipv4 and the ipv6 fields to different values.

```
           +--rw oc-hashing:hash-field-modes
              +--rw oc-hashing:config
              |  +--rw oc-hashing:ipv4?                hash-field-type-ipv4
              |  +--rw oc-hashing:ipv6?                hash-field-type-ipv6
              |  +--rw oc-hashing:transport-ports?     hash-field-type-transport
              |  +--rw oc-hashing:ingress-interface?   boolean
```

Reply:
Thanks, understood. But can one make the `transport-ports` input conditional on whether it's ipv4 vs. ipv6?

Reply:
Ah yes, @sallylsy this makes sense to me if it does for you too?

Reply:
Make sense to me. I will make it an enum for ipv4 and ipv6